files/hello_world_example_app/README.md

```diff
@@ -0,0 +1,36 @@
+This is a hello ROS example app. It subscribes to `/chatter` topic and when a message is received, the following message is published in `/a_chatter`:
+
+    data: "hello world from android ndk __COUNTER_VALUE__"
+
+USAGE
+-------
+
+1. IP addresses are harcoded, so you must edit the master URI and the cellphone ip address in the following file:
+
+        app/src/main/cpp/main.cpp
+
+2. Build the samples with do_everything script.
+
+3. Install the app in your cellphone using adb -d install as explained here:
+        <https://developer.android.com/studio/build/building-cmdline>
+
+4. Execute roscore in a terminal with ros sourced. Remember to export first your ip address:
+        
+        export ROS_IP=__YOUR_IP_ADDRESS__
+
+5. Subscribe to /a_chatter in another terminal with ros setup sourced:
+    
+        rostopic echo /a_chatter
+
+6. Publish to /chatter, in another terminal with ros setup sourced and ROS_IP exported:
+
+        rostopic pub /chatter std_msgs/String "__YOUR_MESSAGE__" -r __PUBLISHING_RATE__
```

Comment:
This is a rather silly comment, but perhaps I'd include a listener inside the app as well which logs to `logcat` directly so as not to require this step for testing purposes.

Reply:
I think that testing both publisher and subscriber is ok.
Maybe, creating a roslunch file for testing is a good idea.